shell/app/node_main.cc

```diff
@@ -167,6 +168,17 @@ int NodeMain(int argc, char* argv[]) {
   }
 #endif
 
+  // Check if parent has passed a preload script.
+  auto* command_line = base::CommandLine::ForCurrentProcess();
+  std::optional<base::FilePath> preload;
+  if (os_env->HasVar("ELECTRON_HAS_NODE_PRELOAD")) {
+    os_env->UnSetVar("ELECTRON_HAS_NODE_PRELOAD");
+    preload = command_line->GetSwitchValuePath(switches::kNodePreload);
+    // Remove the last arg, which is the preload script.
+    // Note that modifying argc only has effects on process.argv in node.
+    argc -= 1;
+  }
```

Comment:
This might break the solution from https://github.com/electron/electron/pull/40579,

```
ELECTRON_RUN_AS_NODE=1 ELECTRON_HAS_NODE_PRELOAD=1 electron.exe --node-preload=<path-to>/evil.js
```

would having check for resources path at every site where the file path is obtained from the cli flag be better ?

```
// At similar sites,
preload = command_line->GetSwitchValuePath(switches::kNodePreload);

if (IsPackaged() && !GetResourcesPath().IsParent(*preload)) {
     LOG(ERROR) << ...;
     // Ignore the preload value 
}
```

Reply:
We don't check suspicious args passed to `ELECTRON_RUN_AS_NODE` process, because it is supposed to run any script and any user can just do:

```
ELECTRON_RUN_AS_NODE=1 electron evil.js
```

Reply:
Ah sorry, this won't break since https://github.com/electron/electron/pull/40579. Not sure why I thought that yesterday but `--node-preload` cannot be passed around via LaunchServices from within sandbox application.